.github/workflows/test.yml

```diff
@@ -116,12 +116,13 @@ jobs:
         path: json.log
         retention-days: 30
 
-    # TODO(APP-3946): restore this test - it is designed to test the ability to change users.
-    # Details: https://github.com/viamrobotics/goutils/pull/245#discussion_r1488649164
-    #
     # this one runs as root for CAP_SETUID
     - name: test run-as-user
-      run: go test -v ./pexec -run TestManagedProcessStart
+      env:
+        TEST_SUBPROC_USER: subproc_user
+      run: |
+        sudo useradd $TEST_SUBPROC_USER
+        sudo go test -v ./pexec -run TestManagedProcessStart
```

Comment:
interesting - now I'm curious why it didn't run when I tried [`sudo -E go test ...`](https://github.com/viamrobotics/goutils/pull/245/commits/49fc12cf66e8e0b8af83ce5b952777525bf25a89#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88L123). I probably commented out an important prior setup step by mistake when I tried that.